XenModel/ServerDBs/StatusReportXmlDocReader.cs

```diff
@@ -128,6 +128,24 @@ private string ParsePropertyName(string p, string table)
             }
         }
 
+        private string SanitizePropertyValue(string value)

```

Comment:
The method SanitizePropertyValue() is inside a loop, thus this method has to be static, otherwise every time it is called, a new instance of Dictionary escapedCharacters will be constructed. It is a lot of memory, I think.
B.T.W., would you think it is a better idea to do the replacement before the loop, something like doc.InnerXml = SanitizePropertyValue(doc.InnerXml); I am not quite sure.

Reply:
Good point about having the escapedCharacters dictionary outside of the function.
I think sanitizing the values is good enough here, as they are the only strings in the XML that would need replacing anyway.